lib/bridge-client/index.js

```diff
@@ -603,12 +604,24 @@ BridgeClient.prototype._shardUploadWorker = function(task, done) {
 BridgeClient.prototype._handleShardStream = function(options) {
   var self = this;
   const {shardStream, index, frame, uploadState, fileHash} = options;
+
+  if (uploadState.file.constructor === Readable) {
+    var tmpName = fileHash
+    var tmpFile = self._store.createWriteStream(fileHash)
+  } 
+
+  if (uploadState.file.constructor === String) {
```

Comment:
file in upload state should not change and is set from the original supplied file (either string or readable stream atm).